### ISSUE TITLE
Feature: Add an optional service provider field to Provider class

### ISSUE THREAD

**Issue by lblackstone** [pulumi/pulumi-kubernetes#695]
Currently, there doesn't seem to be a way to tell what service provider a Provider references (AKS, EKS, GKE, etc.). Since each service provider has subtle differences in behavior, it would be nice to have a built-in way to drive conditional behavior.

**Comment by lukehoban** [comment]
Would this be used for anything?  Or is it effectively just a tag?

Another use case for this would be generating nicer URLs for resources when we know they are associated with a GKE cluster for example - though for that purpose, we probably need more than just "this is GKE", but more "this is this particular GKE cluster identity".

**Comment by lblackstone** [comment]
The initial use case was for selecting which status field in a Service object has the endpoint configuration; this isn't standard across the major clouds.

AFAICT, there's no way to reliably introspect this information from a kubeconfig, but we could set it for clusters created with Pulumi. This could certainly enable richer integration with our dashboard.

**Comment by lukehoban** [comment]
> The initial use case was for selecting which status field in a Service object has the endpoint configuration; this isn't standard across the major clouds.

But is it determined solely by the cloud provider?  Or is it potentially confugurable within the cloud provider?

It seems Kubernetes should have some way to ask/answer these questions that isn't heuristic based on "was it AKS?" style questions?

**Comment by lblackstone** [comment]
Yes, it completely depends on the cloud provider's implementation for provisioning cloud resources. This one is for LoadBalancers, but resources like PVs, Ingress, etc., are also subject to differences.

A strong concept of cluster identity is sorely lacking upstream.
cc @hausdorff
